cpp/dolfinx/fem/dofmapbuilder.cpp

```diff
@@ -438,39 +513,37 @@ std::pair<std::vector<std::int64_t>, std::vector<int>> get_global_indices(
   std::vector<std::vector<int>> disp_recv(index_maps.size());
   for (std::size_t d = 0; d < index_maps.size(); ++d)
   {
-    // FIXME: This should check which dimension are needed by the dofmap
     auto map = index_maps[d];
-    if (map)
-    {
-      std::span src = map->src();
-      std::span dest = map->dest();
-      MPI_Dist_graph_create_adjacent(
-          map->comm(), src.size(), src.data(), MPI_UNWEIGHTED, dest.size(),
-          dest.data(), MPI_UNWEIGHTED, MPI_INFO_NULL, false, &comm[d]);
-
-      // Number and values to send and receive
-      const int num_indices = global[d].size();
-      std::vector<int> size_recv;
-      size_recv.reserve(1); // ensure data is not a nullptr
-      size_recv.resize(src.size());
-      MPI_Neighbor_allgather(&num_indices, 1, MPI_INT, size_recv.data(), 1,
-                             MPI_INT, comm[d]);
-
-      // Compute displacements for data to receive. Last entry has total
-      // number of received items.
-      disp_recv[d].resize(src.size() + 1);
-      std::partial_sum(size_recv.begin(), size_recv.begin() + src.size(),
-                       disp_recv[d].begin() + 1);
-
-      // TODO: use MPI_Ineighbor_alltoallv
-      // Send global index of dofs to neighbors
-      all_dofs_received[d].resize(disp_recv[d].back());
-      MPI_Ineighbor_allgatherv(global[d].data(), global[d].size(), MPI_INT64_T,
-                               all_dofs_received[d].data(), size_recv.data(),
-                               disp_recv[d].data(), MPI_INT64_T, comm[d],
-                               &requests[requests_dim.size()]);
-      requests_dim.push_back(d);
-    }
+    assert(map);
+
+    std::span src = map->src();
+    std::span dest = map->dest();
+    MPI_Dist_graph_create_adjacent(
+        map->comm(), src.size(), src.data(), MPI_UNWEIGHTED, dest.size(),
+        dest.data(), MPI_UNWEIGHTED, MPI_INFO_NULL, false, &comm[d]);
+
+    // Number and values to send and receive
+    const int num_indices = global[d].size();
+    std::vector<int> size_recv;
+    size_recv.reserve(1); // ensure data is not a nullptr
+    size_recv.resize(src.size());
+    MPI_Neighbor_allgather(&num_indices, 1, MPI_INT, size_recv.data(), 1,
+                           MPI_INT, comm[d]);
+
+    // Compute displacements for data to receive. Last entry has total
+    // number of received items.
+    disp_recv[d].resize(src.size() + 1);
+    std::partial_sum(size_recv.begin(), size_recv.begin() + src.size(),
+                     disp_recv[d].begin() + 1);
+
+    // TODO: use MPI_Ineighbor_alltoallv
```

Comment:
Can this be done now?

Reply:
Maybe. In another PR. I'd like to just get this merged.